### ISSUE TITLE
Error with Preact 10 and arrow-function components

### ISSUE THREAD

**Issue by fabiodcorreia** [preactjs/enzyme-adapter-preact-pure#37]
Hi,

If I mount a Class component the tests work fine, but when I try a functional component (stateless in this case) I get the following error

`TypeError: Cannot read property 'render' of undefined`

When I run this code
```javascript
mount(<Input name="input-name" type="text" />)
```
And my component is defined with

```javascript
const Input = (props: Props) => {
    return (
        <div> //....
````
I'm using preact 10 with typescript, it's the first time that I use enzyme so maybe I'm doing something wrong. I also have this configuration that is called at the

```javascript
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-preact-pure';
configure({ adapter: new Adapter() });

export { mount };
```

The mount that I use on my test is the **mount** that gets exported.

**Comment by robertknight** [comment]
Hello,

Using `mount(...)` with functional components + Preact 10 should work. Can you post the full stack trace associated with the error please?

**Comment by fabiodcorreia** [comment]
Here it is, now that I look to the stack trace I see it's using preact10 so that part is correct.

`
TypeError: Cannot read property 'render' of undefined

      30 |     it('test', () => {
    > 31 |         const wrapper = mount(<Input name="input-name" type="text" />);
         |                         ^
      32 |         console.log(wrapper.props());

      at nodeTypeFromType (node_modules/enzyme-adapter-preact-pure/build/src/preact10-rst.js:93:36)
      at rstNodeFromComponent (node_modules/enzyme-adapter-preact-pure/build/src/preact10-rst.js:137:20)
      at rstNodeFromVNode (node_modules/enzyme-adapter-preact-pure/build/src/preact10-rst.js:71:16)
      at node_modules/enzyme-adapter-preact-pure/build/src/preact10-rst.js:58:19
          at Array.flatMap (<anonymous>)
      at rstNodesFromChildren (node_modules/enzyme-adapter-preact-pure/build/src/preact10-rst.js:57:45)
      at rstNodeFromVNode (node_modules/enzyme-adapter-preact-pure/build/src/preact10-rst.js:74:16)
      at MountRenderer.getNode [as _getNode] (node_modules/enzyme-adapter-preact-pure/build/src/preact10-rst.js:170:19)
      at MountRenderer.Object.<anonymous>.MountRenderer.render (node_modules/enzyme-adapter-preact-pure/build/src/MountRenderer.js:41:29)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:134:16)
      at Object.mount (node_modules/enzyme/build/mount.js:21:10)
      at Object.it (test/input.spec.tsx:31:25)

`

**Comment by robertknight** [comment]
I can see the issue. The problem is to do with functional components declared using arrow functions and not transpiled down to ES5.  In the current release of the adapter, `function Component (props) { return ... }` works, but `const Component = (props) => ...` does not. This is a bug.

**Comment by fabiodcorreia** [comment]
If you can give me some lights maybe I can try to fix and make a PR. Otherwise I will wait for the fix.

Btw thanks for the quick feedback.

**Comment by robertknight** [comment]
Thanks for the offer. Since I was already looking at the code to figure out the problem I went ahead and merged a fix. This has been published as v1.13.1. Thanks again for the report!
